### ISSUE TITLE
P8 hosts fail to reboot since 5c76e0c

### ISSUE THREAD

**Issue by AlexanderAmelkin** [openbmc/openbmc#3204]
**[Description]**
Initially the host state is 'running':
```
CurrentHostState = xyz.openbmc_project.State.Host.HostState.Running
RequestedHostTransition = xyz.openbmc_project.State.Host.Transition.On
```
Reboot the host with the following command:
```
busctl set-property xyz.openbmc_project.State.Host \
/xyz/openbmc_project/state/host0 xyz.openbmc_project.State.Host \
RequestedHostTransition s \
xyz.openbmc_project.State.Host.Transition.Reboot
```
The host is expected to start rebooting. In reality it goes into the following state:
```
CurrentHostState = xyz.openbmc_project.State.Host.HostState.Running
RequestedHostTransition = xyz.openbmc_project.State.Host.Transition.Reboot
```
**[Analysis]**

Setting RequestedHostTransition to `xyz.openbmc_project.State.Host.Transition.Reboot` launches the following sequence:

  - `obmc-host-reboot@0.target`
    - `obmc-host-shutdown@0.target`
    - `phosphor-reboot-host@0.service`
      - `obmc-host-startmin@0.target`

Before [this commit](https://github.com/openbmc/openbmc/commit/5c76e0cf9f742cae4156c30c033af50be1135ab7) the last step was `obmc-host-start@0.target`.

Unlike `obmc-host-start@0.target` did, the new `obmc-host-startmin@0.target` doesn't launch `obmc-start-host@0.service` and `op-init-pnor@0.service`.

Service `obmc-start-host@0.service` is required for starting the host (it calls `boot()` method for `org.openbmc.control.Host` interface on `/org/openbmc/control/host0` path).

Hence, the host doesn't start after reboot.

This probably doesn't have effect on P9 (can't check as we don't have them), but for VESNIN, which is P8-based, this ruins reboots.

It is proposed to make `obmc-start-host@0.service` required by `obmc-host-startmin@0.target`, not by `obmc-host-start@0.target`.

**Comment by mine260309** [comment]
This is good finding.

@geissonator When we do changes that may affect host on/off/reboot, we really should have both P8 and P9 systems tested.

I think the correct fix is to make `op-init-pnor@0.service` and `op-start-host@.service` to be required by `obmc-host-startmin@.target`, which is explained in the commit, that
> The startmin target will be the minimum services required to start the host.

On P8 it means  `op-init-pnor@0.service` and `op-start-host@.service` shall be run for this target.

I will test and post a fix.

**Comment by rfrandse** [comment]
<https://gerrit.openbmc-project.xyz/10819> Make op-start-host service required by obmc-host-startmin
Resolves  #3204 P8 hosts fail to reboot since 5c76e0c
